src/internal/connector/exchange/service_iterators.go

```diff
@@ -276,65 +292,92 @@ func getFetchIDFunc(category path.CategoryType) (FetchIDFunc, error) {
 	}
 }
 
+// ---------------------------------------------------------------------------
```

Comment:
while these changes look like they would be quite useful for compacting code (I only skimmed them so no detailed comments. I would recommend trying to get a single interface to use instead of generics if possible), they need to be in a separate PR. Rewriting this logic is not related to reusing delta tokens or populating the proper fields on collections

Reply:
For clarity: are you asking for this to split into two separate PRs, or setting expectations for the future?  I'm good with either.

Reply:
split this PR please

Reply:
done.